provisioner/clusterpy_test.go

```diff
@@ -697,14 +748,16 @@ func TestPerformDeletion(t *testing.T) {
 			var deleted []string
 			for _, action := range client.Actions() {
 				ignore := false
-				for _, ce := range tc.clientErrors {
-					if ce.matches(action) {
-						ignore = true
-						break
+				if tc.clientErrors != nil {
+					for _, ce := range tc.clientErrors(state) {
+						if ce.matches(action) {
+							ignore = true
+							break
+						}
 					}
 				}
 
-				if action, ok := action.(k8stesting.DeleteAction); ok && !ignore {
+				if action, ok := action.(k8stesting.DeleteAction); ok && !ignore && action.GetVerb() == "delete" {
```

Comment:
Why is `&& action.GetVerb() == "delete"` needed now? Is it now implied by `action` being a `k8stesting.DeleteAction` already or did something change?

Reply:
because the `clientReactors` is adding one more action (`k8stesting.PatchAction`) that is handling the annotation

so the if statement would have caught 2 actions and duplicated the output

> is it now implied by action being a k8stesting.DeleteAction already
it was already the case no change here isn't it?

Reply:
I mean, we already check that `action` is a `k8stesting.DeleteAction` (via `ok`). A `k8stesting.PatchAction` should already be filtered out there, shouldn't it?

Reply:
I thought so, but no, it is not filtered out.
because of the hierarchy of the Action interfaces and their implementations, the PatchActionImpl is also implementing DeleteAction. 
```
replace (
	k8s.io/apimachinery => k8s.io/apimachinery v0.21.9
	k8s.io/client-go => k8s.io/client-go v0.21.9
)
```
```go
// You can edit this code!
// Click here and start typing.
package main

import (
	"fmt"

	"k8s.io/apimachinery/pkg/runtime/schema"
	k8stypes "k8s.io/apimachinery/pkg/types"
	k8stesting "k8s.io/client-go/testing"
)

func test(a k8stesting.Action) {
	if _, ok := a.(k8stesting.DeleteAction); ok {
		fmt.Println("Action is a DeleteAction")
	} else {
		fmt.Println("Action is not DeleteAction")
	}
}

func main() {
	gvr := schema.GroupVersionResource{Group: "group", Version: "version", Resource: "namespaces"}
	testAction := k8stesting.NewRootPatchAction(gvr, "fake", k8stypes.JSONPatchType, nil)
	test(testAction)
}
```
![Screenshot 2023-01-17 at 11 05 34](https://user-images.githubusercontent.com/1551247/212869304-0cabea1e-c354-403b-976b-366b2e37ed74.png)

it seems a bug that exists in this version of k8s, I checked the latest version and it have been fixed

Reply:
👍

Reply:
![image](https://user-images.githubusercontent.com/36162/212898630-ed1f0cf6-81b0-4e23-b613-2cc90c9a4be6.png)